soweego/linker/classifiers.py

```diff
@@ -441,3 +396,184 @@ def _create_model(
+
+        kwargs = {**constants.GATED_ENSEMBLE_PARAMS, **kwargs}
+
+        self.num_features = num_features
+        self.num_folds = kwargs.pop('folds', 2)
+        self.meta_layer = kwargs.pop('meta_layer')
+
+        estimators = []
+        for clf in constants.CLASSIFIERS_FOR_ENSEMBLE:
+            model = utils.init_model(
+                clf, num_features=self.num_features, **kwargs
+            )
+
+            estimators.append((clf, model.kernel))
+
+        self.kernel = SuperLearner(verbose=2, n_jobs=1, folds=self.num_folds)
+
+        # use as output the probability of a given class (not just
+        # the class itself)
+        self.kernel.add(estimators, proba=True)
+
+        self.kernel.add_meta(
+            utils.init_model(
+                self.meta_layer, len(estimators) * self.num_folds, **kwargs
+            ).kernel
+        )
+
+    def prob(self, feature_vectors: pd.DataFrame) -> pd.Series:
+        """Classify record pairs and include the probability score
+        of being a match.
+
+        :param feature_vectors: a :class:`DataFrame <pandas.DataFrame>`
+          computed via record pairs comparison. This should be
+          :meth:`recordlinkage.Compare.compute` output.
+          See :func:`extract_features() <soweego.linker.workflow.extract_features>`
+          for more details
+        :return: the classification results
+        """
+        classifications = self.kernel.predict(feature_vectors)
+
+        return pd.Series(classifications, index=feature_vectors.index)
+
+    def __repr__(self):
+        return (
+            f'{self.__class__.__name__}('
+            f'num_folds={self.num_folds}, '
+            f'meta_layer={self.meta_layer}) '
+        )
+
+
+class StackedEnsambleClassifier(SKLearnAdapter, BaseClassifier):
+    """Ensemble of stacked classifiers, meaning that classifiers are arranged in layers
+    with the next layer getting as input the output of the last layer.
+    The predictions of the final layer are merged with a meta-learner (the same happens for
+    ~:class:`soweego.linker.GateEnsambleClassifier`), which decides the final
+    output based on the prediction of the base classifiers.
+    """
+
+    def __init__(self, num_features, **kwargs):
+        super(StackedEnsambleClassifier, self).__init__()
+
+        kwargs = {**constants.STACKED_ENSEMBLE_PARAMS, **kwargs}
+
+        self.num_features = num_features
+        self.num_folds = kwargs.pop('folds', 2)
+        self.meta_layer = kwargs.pop('meta_layer')
+
+        def init_estimators(num_features):
+            estimators = []
+            for clf in constants.CLASSIFIERS_FOR_ENSEMBLE:
+                model = utils.init_model(
+                    clf, num_features=num_features, **kwargs
+                )
+
+                estimators.append((clf, model.kernel))
+            return estimators
+
+        self.kernel = SuperLearner(verbose=2, n_jobs=1, folds=self.num_folds)
```

Comment:
Same comment as per the gate classifier

Reply:
Same answer as given for `GatedEnsembleClassifier` :smile: